cupy/fft/_fft.py

```diff
@@ -209,15 +210,10 @@ def _fft_c2c(a, direction, norm, axes, overwrite_x, plan=None):
 
 def _fft(a, s, axes, norm, direction, value_type='C2C', overwrite_x=False,
          plan=None):
-    if isinstance(a, np.ndarray):
+    if not isinstance(a, cupy.ndarray):
```

Comment:
It never accepted foreign arrays that support the CAI protocol, nor did we write a test for that need.
```python
Traceback (most recent call last):
  File "test_fft.py", line 28, in <module>
    dummy_b = cp.fft.fftn(dummy)
  File "/home/leofang/cupy_cuda110/cupy/fft/_fft.py", line 754, in fftn
    func = _default_fft_func(a, s, axes)
  File "/home/leofang/cupy_cuda110/cupy/fft/_fft.py", line 625, in _default_fft_func
    a.ndim == 1 or not config.enable_nd_planning):
AttributeError: 'DummyObjectWithCudaArrayInterface' object has no attribute 'ndim'
```
In fact my gut feeling is most of CuPy APIs do not directly support CAI. 

The check here was added mainly to prevent numpy arrays from being passed to high-level APIs when a multi-GPU transform is requested.